core/state_transition.go

```diff
@@ -730,6 +743,14 @@ func (st *stateTransition) refundGas(refundQuotient uint64) uint64 {
 		st.evm.Config.Tracer.OnGasChange(st.gasRemaining, st.gasRemaining+refund, tracing.GasChangeTxRefunds)
 	}
 
+	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && !st.msg.IsDepositTx {
+		// Return ETH to transaction sender for operator cost overcharge.
+		operatorCostGasLimit := st.evm.Context.OperatorCostFunc(new(big.Int).SetUint64(st.msg.GasLimit), st.evm.Context.Time)
+		operatorCostGasUsed := st.evm.Context.OperatorCostFunc(new(big.Int).SetUint64(st.gasUsed()), st.evm.Context.Time)
+
+		st.state.AddBalance(st.msg.From, new(uint256.Int).Sub(operatorCostGasLimit, operatorCostGasUsed), tracing.BalanceIncreaseGasReturn)
```

Comment:
Could we add a sanity check in here that panic's if `operatorCostGasUsed > operatorCostGasLimit`. It should never happen but it would create a catastrophic ETH minting bug if it ever did.

Reply:
Also, `OperatorCostFunc` is documented as returning `nil` if not fee is charged but this isn't checking for a `nil` return.

Reply:
Fixed